pallets/creditcoin/src/migrations/v6.rs

```diff
@@ -74,6 +75,7 @@ pub mod tests {
 				to: [0u8; 256].into_bounded(),
 				tx_id: [0u8; 256].into_bounded(),
 				contract: Default::default(),
+				contract_type: ContractType::GCRE,
```

Comment:
Yeah, Alex is right. 

You want to copy the definitions of the changed types from before this PR into the migration here. Then make sure the rest of the migration refers to the copied types (this is usually as easy as removing the import of `crate::types::Whatever` so that things refer to the copied types).

So in this case, you would need to copy the old definitions for `UnverifiedCollectedCoins` and `Task` (since `Task` includes `UnverifiedCollectedCoins` in its definition) into this migration.

This case is a little extra tricky though because of the fact that the write to the migrated storage is abstracted behind the `TaskScheduler` trait (`TaskScheduler::insert`). I think the easiest way to handle this is effectively to just do what that call is doing behind the scenes. So I imagine something like

```rust
// this module is because the address from a storage item is `twox_128(pallet_prefix) ++ twox_128(storage_name)`. We need the storage name to be `PendingTasks` so the address is correct. Since we already have a `PendingTasks` defined, I just stick it in a submodule
mod new {
    
    #[storage_alias]
    pub type PendingTasks<T: Config> = StorageDoubleMap<
    	TaskScheduler, // the prefix for the storage item, which is generally the name of the pallet that defines the storage. We use an identifier instead of a string here because that's what the macro expects.
    	Identity,
    	BlockNumberOf<T>,
    	Identity,
    	TaskId<HashOf<T>>,
    	Task<AccountIdOf<T>, BlockNumberOf<T>, HashOf<T>, MomentOf<T>>, // the `Task` we copied into this migration
    >;
}
```

and then instead of `T::TaskScheduler::insert(...)` you would do

```rust
    new::PendingTasks::<T>::insert(...)
```